File: generator.py
## a code generator for the ALU chain in the 32-bit ALU
## look at example_generator.py for inspiration
##
## python generator.py

from __future__ import print_function

width = 32
for i in range(2, width):
    print("    or o{0}(chain[{0}], in[{0}], chain[{1}]);".format(i, i - 1))

File: generator.cpp
// a code generator for the ALU chain in the 32-bit ALU
// look at example_generator.cpp for inspiration

// make generator
// ./generator

#include <cstdio>
using std::printf;

int
main() {
    int width = 32;
    for (int i = 2 ; i < width ; i ++) {
        printf("    or o%d(chain[%d], in[%d], chain[%d]);\n", i, i, i, i-1);
    }
}
